### DISCUSSION TITLE
[2.*] xmlhttprequest-ssl vuneability

### DISCUSSION THREAD

**Discussion by xandros15** [socketio/socket.io#3906] [Q&A] [answered]
When I try to fix vulnerability I got collision problem, and say I should upgrade socket.io to version 4. 

I cannot do that, because that will break my application. 

Is there any option that vulnerability will be fixed for version 2?

**Comment by darrachequesne** [comment, marked as answer]
This should be fixed now: [`engine.io-client@3.5.2`](https://github.com/socketio/engine.io-client/releases/tag/3.5.2)

Thanks for the heads-up :pray:
